### PR TITLE
Test coverage for GitShellOutStrategy

### PR DESCRIPTION
**Pull request by smashwilson**
**Please be sure to read the [contributor's guide to the GitHub package](https://github.com/atom/github/blob/master/CONTRIBUTING.md) before submitting any pull requests.**

### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* Suggestion: You can use checklists to keep track of progress for the sections on metrics, tests, documentation, and user research.

### Description of the Change

This adds unit tests to `git-strategies.test.js` to cover the [remaining uncovered lines reported by CodeCov](https://codecov.io/gh/atom/github/src/4786390d181001cb97ad7fa20d36fa1fb1ec1e89/lib/git-shell-out-strategy.js).

### Alternate Designs

_N/A_

### Benefits

GitShellOutStrategy is one of the more prevalent culprits in test coverage flapping that we see on unrelated PRs. This is one step toward minimizing those changes and keeping CodeCov output relavent.

### Possible Drawbacks

_N/A_

### Applicable Issues

_N/A_

### Metrics

_N/A_

### Tests

Ideally this should raise [GitShellOutStrategy's coverage](https://codecov.io/gh/atom/github/src/master/lib/git-shell-out-strategy.js) to 100%.

### Documentation

_N/A_

### Release Notes

_N/A_

### User Experience Research (Optional)

_N/A_

### CODE REVIEW

**Comment by smashwilson** [comment]
And 100% coverage for `GitShellOutStrategy` 👌

File: lib/git-shell-out-strategy.js [hunk 1]
```diff
@@ -61,9 +61,9 @@ const DISABLE_COLOR_FLAGS = [
  * Ex: on Mac ~kuychaco/ is expanded to the specified user’s home directory (/Users/kuychaco)
  * Regex translation:
  * ^~ line starts with tilde
- * ([^/]*)/ captures non-forwardslash characters before first slash
+ * ([^\\\\/]*)[\\\\/] captures non-slash characters before first slash
  */
-const EXPAND_TILDE_REGEX = new RegExp('^~([^/]*)/');
+const EXPAND_TILDE_REGEX = new RegExp('^~([^\\\\/]*)[\\\\/]');
```
Comment (annthurium) [inline lib/git-shell-out-strategy.js]: it's so much more...expandy!  (I really wanted to come up with a better joke about this but couldn't.)

File: test/git-strategies.test.js [hunk 2]
```diff
@@ -99,28 +151,50 @@ import * as reporterProxy from '../lib/reporter-proxy';
...
         const nonExistentCommitTemplatePath = path.join(workingDirPath, 'file-that-doesnt-exist');
         await git.setConfig('commit.template', nonExistentCommitTemplatePath);
         await assert.isRejected(
           git.fetchCommitMessageTemplate(),
           `Invalid commit template path set in Git config: ${nonExistentCommitTemplatePath}`,
         );
       });
+
+      it('replaces ~ with your home directory', async function() {
+        // Fun fact: even on Windows, git does not accept "~\does-not-exist.txt"
```
Comment (annthurium) [inline test/git-strategies.test.js]: lol really?

File: lib/git-shell-out-strategy.js [hunk 3]
```diff
@@ -807,11 +816,8 @@ export default class GitShellOutStrategy {
     }
   }
 
-  mergeTrailers(commitMessage, trailers, unfold) {
+  mergeTrailers(commitMessage, trailers) {
     const args = ['interpret-trailers'];
-    if (unfold) {
```
Comment (annthurium) [inline lib/git-shell-out-strategy.js]: good catch on this unused param

File: test/git-strategies.test.js [hunk 4]
```diff
@@ -982,6 +1115,14 @@ import * as reporterProxy from '../lib/reporter-proxy';
         await git.setConfig('awesome.devs', 'BinaryMuse,kuychaco,smashwilson');
         assert.equal('BinaryMuse,kuychaco,smashwilson', await git.getConfig('awesome.devs'));
       });
+
+      it('propagates unexpected git errors', async function() {
+        const workingDirPath = await cloneRepository('three-files');
+        const git = createTestStrategy(workingDirPath);
+        sinon.stub(git, 'exec').rejects(new Error('AHHHH'));
```
Comment (annthurium) [inline test/git-strategies.test.js]: a+ fake error names in this pull request

**Review by annthurium — APPROVED** [review]
woo!

File: test/git-strategies.test.js (hunk 2 above)
Comment (smashwilson) [reply test/git-strategies.test.js]: Yuuuup.

**Pull request merged**